packages/nimble-components/src/table/models/table-layout-manager.ts

```diff
@@ -0,0 +1,255 @@
+import type { TableRecord } from '../types';
+
+/**
+ * This class manages the layout of columns within a Table.
+ * @interal
+ */
+export class TableLayoutManager<TData extends TableRecord> {
+    @observable
+    public isColumnBeingSized = false;
+
+    @observable
+    public activeColumnIndex?: number;
+
+    private activeColumnDivider?: number;
+    private gridSizedColumns?: TableColumn[];
+    private visibleColumns: TableColumn[] = [];
+    private initialTableScrollableWidth?: number;
+    private initialTableScrollableMinWidth?: number;
+    private initialColumnTotalWidth?: number;
+    private currentTotalDelta = 0;
+    private dragStart = 0;
+    private leftColumnIndex?: number;
+    private rightColumnIndex?: number;
+    private initialColumnPixelWidths: {
+        initalColumnFractionalWidth: number,
+        initialPixelWidth: number,
+        minPixelWidth: number
+    }[] = [];
+
+    public constructor(private readonly table: Table<TData>) {}
+
+    public getGridTemplateColumns(): string {
+        return this.getVisibleColumns()
+            .map(column => {
+                const {
+                    minPixelWidth,
+                    currentPixelWidth,
+                    currentFractionalWidth
+                } = column.columnInternals;
+                if (currentPixelWidth) {
+                    const coercedPixelWidth = Math.max(
+                        minPixelWidth,
+                        currentPixelWidth
+                    );
+                    return `${coercedPixelWidth}px`;
+                }
+
+                return `minmax(${minPixelWidth}px, ${currentFractionalWidth}fr)`;
+            })
+            .join(' ');
+    }
+
+    /**
+     * Sets up state related to interactively sizing a column.
+     * @param dragStart The x-position from which a column size was started
+     * @param activeColumnDivider The divider that was clicked on
+     */
+    public beginColumnInteractiveSize(
+        dragStart: number,
+        activeColumnDivider: number
+    ): void {
+        this.activeColumnDivider = activeColumnDivider;
+        this.leftColumnIndex = this.getLeftColumnIndexFromDivider(
+            this.activeColumnDivider
+        );
+        this.rightColumnIndex = this.leftColumnIndex + 1;
+        this.activeColumnIndex = this.leftColumnIndex + (this.activeColumnDivider % 2);
+        this.dragStart = dragStart;
+        this.currentTotalDelta = 0;
+        this.visibleColumns = this.getVisibleColumns();
+        this.setColumnsToFixedSize();
+        this.initialTableScrollableWidth = this.table.viewport.scrollWidth;
+        this.initialTableScrollableMinWidth = this.table.tableScrollableMinWidth;
+        this.initialColumnTotalWidth = this.getTotalColumnFixedWidth();
+        this.isColumnBeingSized = true;
+        document.addEventListener('mousemove', this.onDividerMouseMove);
+        document.addEventListener('mouseup', this.onDividerMouseUp);
+    }
```

Comment:
Have we thought of an intent to cancel the column resize? I found myself wanting to push the `esc` to abort an ongoing resize operation.

Reply:
Might be pretty straightforward to represent under pointer events. i.e. the esc key would just restore the sizes and releasePointerCapture (which fires pointercancel I believe) and leaves you ready for the next pointerdown.

Reply:
I think we can tackle this along with the pointer events I've created [this issue](https://github.com/ni/nimble/issues/1362) for.